fix issue 1700: support mutltiple names for mcpbridge || fix issue 1700: support muttiple names for mcpbridge

Chi-Kai opened the pull request:
<!-- Please make sure you have read and understood the contributing guidelines -->

### Ⅰ. Describe what this PR did
fix issue 1700: support mutltiple names for mcpbridge

### Ⅱ. Does this pull request fix one issue?
<!-- If that, add "fixes #xxx" below in the next line, for example, fixes #97. -->

fixes #1700 

### Ⅲ. Why don't you add test cases (unit test/integration test)? 


### Ⅳ. Describe how to verify it

主要思路: 将多个不同name的mcpbridge聚合为一个reconciler输入，同时移除相关name == default检查。

测试配置:

```yaml
  apiVersion: networking.higress.io/v1
  kind: McpBridge
  metadata:
    name: test-env-a
    namespace: higress-system
  spec:
    registries:
    - type: static
      name: httpbin-service
      domain: httpbin.org
      port: 80
  ---
  apiVersion: networking.higress.io/v1
  kind: McpBridge
  metadata:
    name: test-env-b
    namespace: higress-system
  spec:
    registries:
    - type: static
      name: another-service
      domain: httpbin.org
      port: 443
      protocol: https
```
测试结果:
![图片](https://github.com/user-attachments/assets/d716b069-e56a-4e50-a232-bb1065bddace)


### Ⅴ. Special notes for reviews


<!--This is a translation content dividing line, the content below is generated by machine, please do not modify the content below-->
---
<!-- Please make sure you have read and understand the contributing guidelines -->

### Ⅰ. Describe what this PR did
fix issue 1700: support muttiple names for mcpbridge

### Ⅱ. Does this pull request fix one issue?
<!-- If that, add "fixes #xxx" below in the next line, for example, fixes #97. -->

fixes #1700

### Ⅲ. Why don't you add test cases (unit test/integration test)?


### Ⅳ. Describe how to verify it

Main idea: Aggregate mcpbridges of multiple different names into a reconciler input, and remove the relevant name == default check.

Test configuration:

```yaml
  apiVersion: networking.higress.io/v1
  kind: McpBridge
  metadata:
    name: test-env-a
    namespace: higress-system
  spec:
    registries:
    - type: static
      name: httpbin-service
      domain: httpbin.org
      port: 80
  ---
  apiVersion: networking.higress.io/v1
  kind: McpBridge
  metadata:
    name: test-env-b
    namespace: higress-system
  spec:
    registries:
    - type: static
      name: another-service
      domain: httpbin.org
      port: 443
      protocol: https
```
Test results:
![Picture](https://github.com/user-attachments/assets/d716b069-e56a-4e50-a232-bb1065bddace)


### V. Special notes for reviews

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/higress-group/higress?pullRequest=2515) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/higress-group/higress?pullRequest=2515) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/higress-group/higress?pullRequest=2515) it.</sub>

Chi-Kai committed:
fix issue1700: support multiple names for mcpbridge

- Remove DefaultMcpbridgeName constraint that limited to single "default" McpBridge
- Add mcpBridges tracking map with mutex for concurrent access
- Add aggregateAllMcpBridges() to merge registries from all McpBridges with name-prefixed registry names
- Update ValidateBackendResource to accept any McpBridge name (remove name check)
- Update e2e test destination annotation to use prefixed registry name

fixes #1700

Signed-off-by: chikai <chikai0307@gmail.com>

CH3CHO committed:
Merge branch 'main' into feature/issue1700
